/*
 * This file was automatically generated by EvoSuite
 * Mon May 02 09:39:23 GMT 2022
 */


import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class Solution_189_ESTest extends Solution_189_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      Solution_189 solution_189_0 = new Solution_189();
      // Undeclared exception!
      try { 
        solution_189_0.checkRecord((String) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("Solution_189", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      Solution_189 solution_189_0 = new Solution_189();
      boolean boolean0 = solution_189_0.checkRecord(" =qKAnLLLiJ^r-sw#?");
      assertFalse(boolean0);
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      Solution_189 solution_189_0 = new Solution_189();
      boolean boolean0 = solution_189_0.checkRecord("!AApxMZ]R");
      assertFalse(boolean0);
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      Solution_189 solution_189_0 = new Solution_189();
      boolean boolean0 = solution_189_0.checkRecord("Q<");
      assertTrue(boolean0);
  }
}
